### PR TITLE
Preview Append

### PR DESCRIPTION
**Pull request by philippfromme**
brave_LhiZ5ATZjp

## Changes

* context pad
  * preview append on hover
* renderer
  * `fill`, `stroke`, `width` and `height` can be overridden
  * no other attributes can be overridden
  * public API only exposes renderers for actual BPMN elements

## Follow-Ups

* https://github.com/bpmn-io/diagram-js/issues/816

---

Closes https://github.com/bpmn-io/bpmn-js/issues/1990

### CODE REVIEW

**Comment by philippfromme** [comment]
@barmac @marstamm I've added back the opacity and updated the changelog.

File: test/spec/features/append-preview/AppendPreviewSpec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    var startEvent = elementRegistry.get('StartEvent_1');
+
+    // when
+    appendPreview.create(startEvent, 'bpmn:Task');
+
+    // assume
+    expect(canvas.getLayer('complex-preview')).to.exist;
+    expect(domQueryAll('.djs-dragger', canvas.getLayer('complex-preview'))).to.have.length(2);
+
+    // whem
```
Comment (barmac) [inline test/spec/features/append-preview/AppendPreviewSpec.js:49]:
```suggestion
    // when
```

**Review by barmac — COMMENTED** [review]
Should we mark BpmnRenderer changes as breaking? We are removing parts of the public API.
Other than that it looks good to me. I love how it improves the UX of the editor.

**Review by marstamm — COMMENTED** [review]
So Subprocess and Call activity are rendered with the same border width. Is this an oversight or do we have the same issues with this approach like we had when I tried to rework the renderer?

image

**Comment by barmac** [comment]
> Is this an oversight or do we have the same issues with this approach like we had when I tried to rework the renderer?

Looks like a bug to me! :(

File: lib/draw/BpmnRenderer.js [hunk 2]
```diff
@@ -852,445 +719,640 @@ export default function BpmnRenderer(
...
+  function renderSubProcess(parentGfx, element, attrs = {}) {
+    var expanded = isExpanded(element);
 
-      /* inner path */ drawRect(parentGfx, element.width, element.height, TASK_BORDER_RADIUS - 3, INNER_OUTER_DIST, innerAttrs);
+    if (!expanded) {
+      attrs = pickAttrs(attrs, [
+        'fill',
+        'stroke'
+      ]);
+    }
```
Comment (marstamm) [inline lib/draw/BpmnRenderer.js:1212]: `strokeWidth` is passed as an attribute into `renderSubProcess`, but it is not picked from the attributes here

**Review by marstamm — CHANGES_REQUESTED** [review]

**Comment by barmac** [comment]
Business Rule Task is broken too:

image

Current:

image

File: lib/draw/BpmnRenderer.js (hunk 2 above)
Comment (philippfromme) [reply lib/draw/BpmnRenderer.js:1212]: I broke this last minute. 🙈 Makes me think I should try adding a snapshot test.

**Comment by philippfromme** [comment]
* added breaking changes to changelog
* fixed call activity stroke width
* fixed sequential marker position
* fixed business rule task icon

Ready for another review.

**Review by barmac — APPROVED** [review]

**Review by marstamm — APPROVED** [review]

**Comment by barmac** [comment]
Let's release it! 🚀

**Pull request merged**